scripts/solve_network.py

```diff
@@ -346,6 +346,116 @@ def add_battery_constraints(n):
     define_constraints(n, lhs, "=", 0, "Link", "charger_ratio")
 
 
+def add_RES_constraints(n, res_share):
+    lgrouper = n.loads.bus.map(n.buses.country)
+    ggrouper = n.generators.bus.map(n.buses.country)
+    sgrouper = n.storage_units.bus.map(n.buses.country)
+    cgrouper = n.links.bus0.map(n.buses.country)
+
+    load = (
+        n.snapshot_weightings.generators
+        @ n.loads_t.p_set.groupby(lgrouper, axis=1).sum()
+    )
+
+    rhs = res_share * load
+
+    res_techs = [
+        "solar",
+        "onwind",
+        "offwind-dc",
+        "offwind-ac",
+        "battery",
```

Comment:
why battery? should be excluded here similar to hydrogen?

Reply:
hydrogen does not appear but you are right. Anyway this should not be hard-coded but taken from config file anyways, and if too much appears it does not matter because there is no "battery" carrier in the generators anyway. If there is, something else is already broken...